src/Abstractions/TaskFailureDetails.cs

```diff
@@ -107,16 +109,78 @@ public bool IsCausedBy(Type targetBaseExceptionType)
     public static TaskFailureDetails FromException(Exception exception)
     {
         Check.NotNull(exception);
+        return FromExceptionRecursive(exception);
+    }
+
+    [return: NotNullIfNotNull(nameof(exception))]
+    static TaskFailureDetails? FromExceptionRecursive(Exception? exception)
+    {
+        if (exception is null)
+        {
+            return null;
+        }
+
         if (exception is CoreOrchestrationException coreEx)
         {
             return new TaskFailureDetails(
                 coreEx.FailureDetails?.ErrorType ?? "(unknown)",
                 coreEx.FailureDetails?.ErrorMessage ?? "(unknown)",
                 coreEx.FailureDetails?.StackTrace,
-                null /* InnerFailure */);
+                FromCoreFailureDetailsRecursive(coreEx.FailureDetails?.InnerFailure) ?? FromExceptionRecursive(coreEx.InnerException));
+        }
+
+        return new TaskFailureDetails(
+            exception.GetType().ToString(),
+            exception.Message,
+            exception.StackTrace,
```

Comment:
I am a bit concerned that stack trace will bloat the size of this payload. Should we consider this now or something to think about later?

Reply:
Would it make sense to have some kind of configuration flag to exclude the stack trace?

Reply:
Probably yeah. Or we could opt to not include stack trace at all for inner exceptions?

Reply:
I think there's an argument to be made that, just like we claim inputs and outputs should be small (and we don't tamper with that payload on the user's behalf) that if there's a very large exception trace then it's on the user to trim it / be aware of the perf cost.

I worry that by trying to optimize this case, we'll just get more customer questions about exception serialization is "incomplete". My recommendation would be to include the full stack trace by default, and to possibly add the feature flag to disable it if we find that, in practice, this is a problem.

Reply:
Let's go with what we have for now, but happy to follow up in a separate set of PRs.